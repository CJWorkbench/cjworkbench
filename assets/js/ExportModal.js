import React from 'react'
import PropTypes from 'prop-types'
import CopyToClipboard from 'react-copy-to-clipboard'
import { logUserEvent } from './utils'
import { Modal, ModalHeader, ModalBody, ModalFooter } from './components/Modal'
import { Trans } from '@lingui/macro'

export default class ExportModal extends React.PureComponent {
  static propTypes = {
    open: PropTypes.bool.isRequired,
    wfModuleId: PropTypes.number.isRequired, // to build download URLs
    toggle: PropTypes.func.isRequired
  }

  state = {
    csvCopied: false,
    jsonCopied: false
  }

  buildUrlString (ext) {
    const path = `/public/moduledata/live/${this.props.wfModuleId}.${ext}`
    if (window.location.href === 'about:blank') {
      // allowing an out for testing (there is no window.location.href during test)
      return path
    } else {
      return new URL(path, window.location.href).href
    }
  }

  get csvUrlString () {
    return this.buildUrlString('csv')
  }

  get jsonUrlString () {
    return this.buildUrlString('json')
  }

  handleCopyCsv = () => {
    this.setState({ csvCopied: true })
  }

  handleLeaveCsv = () => {
    this.setState({ csvCopied: false })
  }

  handleCopyJson = () => {
    this.setState({ jsonCopied: true })
  }

  handleLeaveJson = () => {
    this.setState({ jsonCopied: false })
  }

  logExport (type) {
    logUserEvent('Export ' + type)
  }

  renderCsvCopyLink () {
    if (this.state.csvCopied) {
      return (
        <div className='clipboard copied' onMouseLeave={this.handleLeaveCsv}>
          <Trans id='js.ExportModal.csvCopyLink.copiedToClipboard' description='This should be all-caps for styling reasons'>
                CSV LINK COPIED TO CLIPBOARD
          </Trans>
        </div>
      )
    } else {
      return (
        <CopyToClipboard text={this.csvUrlString} onCopy={this.handleCopyCsv} className='clipboard test-csv-copy'>
          <div>
            <Trans id='js.ExportModal.csvCopyLink.copyLiveLink' description='This should be all-caps for styling reasons'>
                COPY LIVE LINK
            </Trans>
          </div>
        </CopyToClipboard>
      )
    }
  }

  renderJsonCopyLink () {
    if (this.state.jsonCopied) {
      return (
        <div className='clipboard copied' onMouseLeave={this.handleLeaveJson}>
          <Trans id='js.ExportModal.jsonCopyLink.copiedToClipboard' description='This should be all-caps for styling reasons'>
                JSON FEED LINK COPIED TO CLIPBOARD
          </Trans>
        </div>
      )
    } else {
      return (
        <CopyToClipboard text={this.jsonUrlString} onCopy={this.handleCopyJson} className='clipboard test-json-copy'>
          <div>
            <Trans id='js.ExportModal.jsonCopyLink.copyLiveLink' description='This should be all-caps for styling reasons'>
                COPY LIVE LINK
            </Trans>
          </div>
        </CopyToClipboard>
      )
    }
  }

  handleClickClose = () => this.props.toggle()

  render () {
    const csvCopyLink = this.renderCsvCopyLink()
    const jsonCopyLink = this.renderJsonCopyLink()

    return (
      <Modal isOpen={this.props.open} className={this.props.className} toggle={this.props.toggle}>
        <ModalHeader>
          <Trans id='js.ExportModal.header.title' description='This should be all-caps for styling reasons'>
                EXPORT DATA
          </Trans>
        </ModalHeader>
        <ModalBody>
          <div className='d-flex justify-content-between flex-row'>
            <div className='dl-file'>
              <Trans id='js.ExportModal.type.CSV' description='This should be all-caps for styling reasons'>
                    CSV
              </Trans>
            </div>
            {csvCopyLink}
          </div>
          <div className='d-flex justify-content-between flex-row mb-3'>
            <input type='url' className='url-link test-csv-field' value={this.csvUrlString} readOnly />
            <div className='download-icon-box'>
              <a
                href={this.csvUrlString} onClick={() => this.logExport('CSV')}
                className='icon-download t-d-gray button-icon test-csv-download' download
              />
            </div>
          </div>
          <div className='d-flex justify-content-between flex-row'>
            <div className='dl-file'>
              <Trans id='js.ExportModal.type.JSON' description='This should be all-caps for styling reasons'>
                    JSON FEED
              </Trans>
            </div>
            {jsonCopyLink}
          </div>
          <div className='d-flex justify-content-between flex-row'>
            <input type='url' className='url-link test-json-field' value={this.jsonUrlString} readOnly />
            <div className='download-icon-box'>
              <a
                href={this.jsonUrlString} onClick={() => this.logExport('JSON')}
                className='icon-download t-d-gray button-icon test-json-download' download
              />
            </div>
          </div>
        </ModalBody>
        <ModalFooter>
          <button type='button' onClick={this.handleClickClose} className='button-blue action-button test-done-button'>
            <Trans id='js.ExportModal.footer.doneButton'>Done</Trans>
          </button>
        </ModalFooter>
      </Modal>
    )
  }
}
