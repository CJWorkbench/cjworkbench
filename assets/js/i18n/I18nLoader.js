import React from 'react'
import { I18nProvider } from '@lingui/react'
import fetchCatalog from './catalogs'

export class I18nLoader extends React.Component {
  state = {
    catalogs: {},
  }
  
  loadCatalog = (language) => {
      this.setState(state => ({
          catalogs: {
            ...state.catalogs,
            [language]: fetchCatalog(language)
          }
      }))
  }
  
  componentDidMount() {
    this.loadCatalog(this.props.language)
  }

  shouldComponentUpdate(nextProps, nextState) {
    const { language } = nextProps
    const { catalogs } = nextState

    if (language !== this.props.language && !catalogs[language]) {
      this.loadCatalog(language)
      return false
    }

    return true
  }

  render () {
    const { children, language } = this.props
    const { catalogs } = this.state

    // Skip rendering when catalog isn't loaded.
    if (!catalogs[language]) {
        return null
    }

    return (
      <I18nProvider language={language} catalogs={catalogs}>
        {children}
      </I18nProvider>
    )
  }
}
