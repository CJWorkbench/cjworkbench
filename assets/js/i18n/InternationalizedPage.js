import React from 'react'
import PropTypes from 'prop-types'
import { en, el } from 'make-plural/plurals'
import { i18n } from '@lingui/core'
import { I18nProvider } from '@lingui/react'
import { currentLocaleId } from './locales'
import Spinner from '../Spinner'

__webpack_public_path__ = window.STATIC_URL + 'bundles/' // eslint-disable-line

async function loadLocale (localeId) {
  const { messages } = await import(
    /* webpackChunkName: "[request]" */
    `@lingui/loader!../../locale/${localeId}/messages.po`
  )
  i18n.loadLocaleData({
    en: { plurals: en },
    el: { plurals: el }
  })
  i18n.load(localeId, messages)
  i18n.activate(localeId)
  return i18n
}

function useActivatedI18n (localeId) {
  const [i18n, setI18n] = React.useState(null)

  React.useEffect(() => {
    loadLocale(currentLocaleId).then(setI18n)
  }, [])

  return i18n
}

export default function InternationalizedPage (props) {
  const { children } = props
  const i18n = useActivatedI18n(currentLocaleId)

  if (i18n === null) {
    return <Spinner />
  }

  return (
    <I18nProvider i18n={i18n} forceRenderOnLocaleChange={false}>
      {children}
    </I18nProvider>
  )
}
InternationalizedPage.propTypes = {
  children: PropTypes.node.isRequired
}
