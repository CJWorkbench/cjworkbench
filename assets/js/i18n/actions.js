const SET_LOCALE = 'SET_LOCALE';

function reduceSetLocale(state, action){
  return {
    ...state,
    locale: action.locale
  };
}

export function setLocaleAction(locale){
  return {
    type: SET_LOCALE,
    locale: locale
  }
}

export const reducerFunctions = {
  [SET_LOCALE]: reduceSetLocale
}

export function localeReducer(state, action){
  if (action.type in reducerFunctions) {
    // Run a registered reducer
    return reducerFunctions[action.type](state, action)
  }
  
  return state;
}
