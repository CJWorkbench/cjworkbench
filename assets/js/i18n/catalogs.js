import enCatalog from '../../locale/en/messages.po'
import elCatalog from '../../locale/el/messages.po'

const catalogs = {
  en: enCatalog,
  el: elCatalog
}

export default function fetchCatalog (language) {
  return catalogs[language]
}
