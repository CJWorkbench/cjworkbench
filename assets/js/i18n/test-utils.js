import { I18nProvider } from '@lingui/react'
import React from 'react'
import { shape, object } from 'prop-types'
import { mount, shallow } from 'enzyme'

/*
 * Below lie utils for testing with i18n.
 * See https://lingui.js.org/guides/testing.html
 */
// Create the I18nProvider to retrieve context for wrapping around.
const intlProvider = new I18nProvider({
  language: 'en',
  catalogs: {
    en: {}
  }
}, {})

const {
  linguiPublisher: {
    i18n: originalI18n
  }
} = intlProvider.getChildContext()

// You customize the i18n object here:
export const i18n = {
  ...originalI18n,
  // provide _ macro, for trying to unwrap default/id or just passing down the key
  _: key => {
    if (typeof key.defaults === 'string') return key.defaults
    if (typeof key.id === 'string') return key.id
    return key
  }
}

/**
 * When using Lingui `withI18n` on components, props.i18n is required.
 */
function nodeWithI18nProp (node) {
  return React.cloneElement(node, { i18n })
}

/*
 * Methods to use
 */
export function shallowWithI18n (node, { context } = {}) {
  return shallow(
    nodeWithI18nProp(node),
    {
      context: Object.assign({}, context, { i18n })
    }
  )
}

export function mountWithI18n (node, { context, childContextTypes } = {}) {
  const newContext = Object.assign({}, context, { linguiPublisher: { i18n } })
  /*
   * I18nProvider sets the linguiPublisher in the context for withI18n to get
   * the i18n object from.
   */
  const newChildContextTypes = Object.assign({},
    {
      linguiPublisher: shape({
        i18n: object.isRequired
      }).isRequired
    },
    childContextTypes
  )
  return mount(
    nodeWithI18nProp(node),
    {
      context: newContext,
      childContextTypes: newChildContextTypes
    }
  )
}
