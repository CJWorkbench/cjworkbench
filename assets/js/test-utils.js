/* globals jest */
import { createStore, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import promiseMiddleware from 'redux-promise-middleware'
import errorMiddleware from './error-middleware'
import { workflowReducer } from './workflow-reducer'

// Returns new mock function that returns given json. Used for mocking "get" API calls
export function jsonResponseMock (json) {
  return jest.fn().mockImplementation(() =>
    Promise.resolve(json)
  )
}

// Returns new mock function that gives an OK HTTP response. Use for mocking "set" API calls
export function okResponseMock () {
  return jsonResponseMock(null)
}

// Helper function to return a promise that resolves after all other promise mocks,
// even if they are chained like Promise.resolve().then(...).then(...)
// Technically: this is designed to resolve on the next macrotask
// https://stackoverflow.com/questions/25915634/difference-between-microtask-and-macrotask-within-an-event-loop-context
export function tick () {
  return new Promise(resolve => {
    setTimeout(resolve, 0)
  })
}

export function sleep (ms) {
  return new Promise(resolve => {
    setTimeout(resolve, ms)
  })
}

/**
 * Return `[ notify, promise ]`; `promise` completes when `notify` is called.
 *
 * Usage:
 *
 *     const [ notify, condition ] = createConditionVariable
 *     const done = condition.then(do_something)
 *     await Promise.all([sleep(0.1).then(notify), done])
 */
export function createConditionVariable () {
  let notify
  const promise = new Promise(resolve => { notify = resolve })
  return [notify, promise]
}

/**
 * Mock Redux store with in-production reducers.
 *
 * This is for integration-style tests: they test the reducer _and_ the caller
 * at the same time.
 *
 * Usage:
 *
 *     const api = { undo: jest.fn() }
 *     const store = mockStoreWithReducer({'workflow': ..., ...}, api)
 *     store.dispatch(Actions.undo())
 *     expect(api.undo).toHaveBeenCalled()
 *     expect(store.getState().workflow).toEqual(...)
 */
export function mockStore (initialState, api = {}) {
  const middlewares = [errorMiddleware(), promiseMiddleware, thunk.withExtraArgument(api)]
  const store = createStore(workflowReducer, initialState, applyMiddleware(...middlewares))
  return store
}
