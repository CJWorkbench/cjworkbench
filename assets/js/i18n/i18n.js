import { setupI18n } from "@lingui/core"
import { currentLocale } from './locales'
import { fetchCatalog } from './catalogs'

export default const i18n = setupI18n({
   language: currentLocale,
   catalogs: {
       currentLocale: fetchCatalog(currentLocale)
   }
})
