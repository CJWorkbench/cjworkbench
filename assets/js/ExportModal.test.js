/* globals afterEach, beforeEach, describe, expect, it, jest */
import React from 'react'
import ExportModal from './ExportModal'
import { mountWithI18n } from './i18n/test-utils'

describe('ExportModal', () => {
  let wrapper

  beforeEach(() => {
    // mount not shallow as we are looking for components down in the tree, e.g. the input fields inside FormGroup
    wrapper = mountWithI18n(
      <ExportModal
        wfModuleId={415}
        open
        toggle={jest.fn()}
        className='menu-test-class'
      />
    )
  })

  afterEach(() => wrapper.unmount())

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('closes', () => {
    wrapper.find('button.test-done-button').simulate('click')
    expect(wrapper.prop('toggle')).toHaveBeenCalled()
  })

  it('should render download links', () => {
    const csvField = wrapper.find('input.test-csv-field')
    expect(csvField.length).toBe(1)
    expect(csvField.props().value).toBe('http://localhost/public/moduledata/live/415.csv')

    const jsonField = wrapper.find('input.test-json-field')
    expect(jsonField.length).toBe(1)
    expect(jsonField.props().value).toBe('http://localhost/public/moduledata/live/415.json')
  })

  it('renders copy to clipboard buttons', () => {
    const csvCopy = wrapper.find('div.test-csv-copy')
    expect(csvCopy).toHaveLength(1)

    const jsonCopy = wrapper.find('div.test-json-copy')
    expect(jsonCopy).toHaveLength(1)
  })

  it('Renders modal links which can be downloaded', () => {
    const csvDownload = wrapper.find('a.test-csv-download')
    expect(csvDownload.prop('href')).toBe('http://localhost/public/moduledata/live/415.csv')

    const jsonDownload = wrapper.find('a.test-json-download')
    expect(jsonDownload.prop('href')).toBe('http://localhost/public/moduledata/live/415.json')
  })
})
