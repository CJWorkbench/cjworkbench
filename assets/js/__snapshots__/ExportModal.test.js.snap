// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportModal should match snapshot 1`] = `
<ExportModal
  className="menu-test-class"
  i18n={
    Object {
      "_": [Function],
      "_activeLanguageData": Object {
        "plurals": [Function],
      },
      "_activeMessages": Object {},
      "_catalogs": Object {
        "en": Object {
          "languageData": Object {},
          "messages": Object {},
        },
      },
      "_dev": Object {
        "compile": [Function],
        "loadLanguageData": [Function],
      },
      "_language": "en",
      "_locales": undefined,
      "_missing": undefined,
      "plural": [Function],
      "select": [Function],
      "selectOrdinal": [Function],
      "t": [Function],
    }
  }
  open={true}
  toggle={[MockFunction]}
  wfModuleId={415}
>
  <Modal
    className="menu-test-class"
    isOpen={true}
    toggle={[MockFunction]}
  >
    <OpenModal
      className="menu-test-class"
      toggle={[MockFunction]}
    >
      <Portal
        containerInfo={
          <body
            class="modal-open"
          >
            <div
              class="modal-backdrop"
            />
            <div
              class="modal menu-test-class"
              role="dialog"
              tabindex="-1"
            >
              <div
                class="modal-dialog"
                role="document"
              >
                <div
                  class="modal-content"
                >
                  <div
                    class="modal-header"
                  >
                    <h5
                      class="modal-title"
                    >
                      js.ExportModal.header.title
                    </h5>
                    <button
                      aria-label="Close"
                      class="close"
                      type="button"
                    >
                      <span
                        aria-hidden="true"
                      >
                        ×
                      </span>
                    </button>
                  </div>
                  <div
                    class="modal-body"
                  >
                    <div
                      class="d-flex justify-content-between flex-row"
                    >
                      <div
                        class="dl-file"
                      >
                        js.ExportModal.type.CSV
                      </div>
                      <div
                        class="clipboard test-csv-copy"
                      >
                        js.ExportModal.csvCopyLink.copyLiveLink
                      </div>
                    </div>
                    <div
                      class="d-flex justify-content-between flex-row mb-3"
                    >
                      <input
                        class="url-link test-csv-field"
                        readonly=""
                        type="url"
                        value="http://localhost/public/moduledata/live/415.csv"
                      />
                      <div
                        class="download-icon-box"
                      >
                        <a
                          class="icon-download t-d-gray button-icon test-csv-download"
                          download=""
                          href="http://localhost/public/moduledata/live/415.csv"
                        />
                      </div>
                    </div>
                    <div
                      class="d-flex justify-content-between flex-row"
                    >
                      <div
                        class="dl-file"
                      >
                        js.ExportModal.type.JSON
                      </div>
                      <div
                        class="clipboard test-json-copy"
                      >
                        js.ExportModal.jsonCopyLink.copyLiveLink
                      </div>
                    </div>
                    <div
                      class="d-flex justify-content-between flex-row"
                    >
                      <input
                        class="url-link test-json-field"
                        readonly=""
                        type="url"
                        value="http://localhost/public/moduledata/live/415.json"
                      />
                      <div
                        class="download-icon-box"
                      >
                        <a
                          class="icon-download t-d-gray button-icon test-json-download"
                          download=""
                          href="http://localhost/public/moduledata/live/415.json"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="modal-footer"
                  >
                    <button
                      class="button-blue action-button test-done-button"
                      type="button"
                    >
                      js.ExportModal.footer.doneButton
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </body>
        }
      >
        <div
          className="modal-backdrop"
        />
        <div
          className="modal menu-test-class"
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="dialog"
          tabIndex="-1"
        >
          <div
            className="modal-dialog"
            role="document"
          >
            <div
              className="modal-content"
            >
              <ModalHeader>
                <div
                  className="modal-header"
                >
                  <h5
                    className="modal-title"
                  >
                    <WithI18n
                      defaults="EXPORT DATA"
                      description="This should be all-caps for styling reasons"
                      id="js.ExportModal.header.title"
                    >
                      <I18n
                        update={true}
                        withHash={true}
                      >
                        <Trans
                          defaults="EXPORT DATA"
                          description="This should be all-caps for styling reasons"
                          i18n={
                            Object {
                              "_": [Function],
                              "_activeLanguageData": Object {
                                "plurals": [Function],
                              },
                              "_activeMessages": Object {},
                              "_catalogs": Object {
                                "en": Object {
                                  "languageData": Object {},
                                  "messages": Object {},
                                },
                              },
                              "_dev": Object {
                                "compile": [Function],
                                "loadLanguageData": [Function],
                              },
                              "_language": "en",
                              "_locales": undefined,
                              "_missing": undefined,
                              "plural": [Function],
                              "select": [Function],
                              "selectOrdinal": [Function],
                              "t": [Function],
                            }
                          }
                          id="js.ExportModal.header.title"
                        >
                          <Render
                            value="js.ExportModal.header.title"
                          >
                            js.ExportModal.header.title
                          </Render>
                        </Trans>
                      </I18n>
                    </WithI18n>
                  </h5>
                  <button
                    aria-label="Close"
                    className="close"
                    onClick={[MockFunction]}
                    type="button"
                  >
                    <span
                      aria-hidden="true"
                    >
                      ×
                    </span>
                  </button>
                </div>
              </ModalHeader>
              <ModalBody>
                <div
                  className="modal-body"
                >
                  <div
                    className="d-flex justify-content-between flex-row"
                  >
                    <div
                      className="dl-file"
                    >
                      <WithI18n
                        defaults="CSV"
                        description="This should be all-caps for styling reasons"
                        id="js.ExportModal.type.CSV"
                      >
                        <I18n
                          update={true}
                          withHash={true}
                        >
                          <Trans
                            defaults="CSV"
                            description="This should be all-caps for styling reasons"
                            i18n={
                              Object {
                                "_": [Function],
                                "_activeLanguageData": Object {
                                  "plurals": [Function],
                                },
                                "_activeMessages": Object {},
                                "_catalogs": Object {
                                  "en": Object {
                                    "languageData": Object {},
                                    "messages": Object {},
                                  },
                                },
                                "_dev": Object {
                                  "compile": [Function],
                                  "loadLanguageData": [Function],
                                },
                                "_language": "en",
                                "_locales": undefined,
                                "_missing": undefined,
                                "plural": [Function],
                                "select": [Function],
                                "selectOrdinal": [Function],
                                "t": [Function],
                              }
                            }
                            id="js.ExportModal.type.CSV"
                          >
                            <Render
                              value="js.ExportModal.type.CSV"
                            >
                              js.ExportModal.type.CSV
                            </Render>
                          </Trans>
                        </I18n>
                      </WithI18n>
                    </div>
                    <CopyToClipboard
                      className="clipboard test-csv-copy"
                      onCopy={[Function]}
                      text="http://localhost/public/moduledata/live/415.csv"
                    >
                      <div
                        className="clipboard test-csv-copy"
                        onClick={[Function]}
                      >
                        <WithI18n
                          defaults="COPY LIVE LINK"
                          description="This should be all-caps for styling reasons"
                          id="js.ExportModal.csvCopyLink.copyLiveLink"
                        >
                          <I18n
                            update={true}
                            withHash={true}
                          >
                            <Trans
                              defaults="COPY LIVE LINK"
                              description="This should be all-caps for styling reasons"
                              i18n={
                                Object {
                                  "_": [Function],
                                  "_activeLanguageData": Object {
                                    "plurals": [Function],
                                  },
                                  "_activeMessages": Object {},
                                  "_catalogs": Object {
                                    "en": Object {
                                      "languageData": Object {},
                                      "messages": Object {},
                                    },
                                  },
                                  "_dev": Object {
                                    "compile": [Function],
                                    "loadLanguageData": [Function],
                                  },
                                  "_language": "en",
                                  "_locales": undefined,
                                  "_missing": undefined,
                                  "plural": [Function],
                                  "select": [Function],
                                  "selectOrdinal": [Function],
                                  "t": [Function],
                                }
                              }
                              id="js.ExportModal.csvCopyLink.copyLiveLink"
                            >
                              <Render
                                value="js.ExportModal.csvCopyLink.copyLiveLink"
                              >
                                js.ExportModal.csvCopyLink.copyLiveLink
                              </Render>
                            </Trans>
                          </I18n>
                        </WithI18n>
                      </div>
                    </CopyToClipboard>
                  </div>
                  <div
                    className="d-flex justify-content-between flex-row mb-3"
                  >
                    <input
                      className="url-link test-csv-field"
                      readOnly={true}
                      type="url"
                      value="http://localhost/public/moduledata/live/415.csv"
                    />
                    <div
                      className="download-icon-box"
                    >
                      <a
                        className="icon-download t-d-gray button-icon test-csv-download"
                        download={true}
                        href="http://localhost/public/moduledata/live/415.csv"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                  <div
                    className="d-flex justify-content-between flex-row"
                  >
                    <div
                      className="dl-file"
                    >
                      <WithI18n
                        defaults="JSON FEED"
                        description="This should be all-caps for styling reasons"
                        id="js.ExportModal.type.JSON"
                      >
                        <I18n
                          update={true}
                          withHash={true}
                        >
                          <Trans
                            defaults="JSON FEED"
                            description="This should be all-caps for styling reasons"
                            i18n={
                              Object {
                                "_": [Function],
                                "_activeLanguageData": Object {
                                  "plurals": [Function],
                                },
                                "_activeMessages": Object {},
                                "_catalogs": Object {
                                  "en": Object {
                                    "languageData": Object {},
                                    "messages": Object {},
                                  },
                                },
                                "_dev": Object {
                                  "compile": [Function],
                                  "loadLanguageData": [Function],
                                },
                                "_language": "en",
                                "_locales": undefined,
                                "_missing": undefined,
                                "plural": [Function],
                                "select": [Function],
                                "selectOrdinal": [Function],
                                "t": [Function],
                              }
                            }
                            id="js.ExportModal.type.JSON"
                          >
                            <Render
                              value="js.ExportModal.type.JSON"
                            >
                              js.ExportModal.type.JSON
                            </Render>
                          </Trans>
                        </I18n>
                      </WithI18n>
                    </div>
                    <CopyToClipboard
                      className="clipboard test-json-copy"
                      onCopy={[Function]}
                      text="http://localhost/public/moduledata/live/415.json"
                    >
                      <div
                        className="clipboard test-json-copy"
                        onClick={[Function]}
                      >
                        <WithI18n
                          defaults="COPY LIVE LINK"
                          description="This should be all-caps for styling reasons"
                          id="js.ExportModal.jsonCopyLink.copyLiveLink"
                        >
                          <I18n
                            update={true}
                            withHash={true}
                          >
                            <Trans
                              defaults="COPY LIVE LINK"
                              description="This should be all-caps for styling reasons"
                              i18n={
                                Object {
                                  "_": [Function],
                                  "_activeLanguageData": Object {
                                    "plurals": [Function],
                                  },
                                  "_activeMessages": Object {},
                                  "_catalogs": Object {
                                    "en": Object {
                                      "languageData": Object {},
                                      "messages": Object {},
                                    },
                                  },
                                  "_dev": Object {
                                    "compile": [Function],
                                    "loadLanguageData": [Function],
                                  },
                                  "_language": "en",
                                  "_locales": undefined,
                                  "_missing": undefined,
                                  "plural": [Function],
                                  "select": [Function],
                                  "selectOrdinal": [Function],
                                  "t": [Function],
                                }
                              }
                              id="js.ExportModal.jsonCopyLink.copyLiveLink"
                            >
                              <Render
                                value="js.ExportModal.jsonCopyLink.copyLiveLink"
                              >
                                js.ExportModal.jsonCopyLink.copyLiveLink
                              </Render>
                            </Trans>
                          </I18n>
                        </WithI18n>
                      </div>
                    </CopyToClipboard>
                  </div>
                  <div
                    className="d-flex justify-content-between flex-row"
                  >
                    <input
                      className="url-link test-json-field"
                      readOnly={true}
                      type="url"
                      value="http://localhost/public/moduledata/live/415.json"
                    />
                    <div
                      className="download-icon-box"
                    >
                      <a
                        className="icon-download t-d-gray button-icon test-json-download"
                        download={true}
                        href="http://localhost/public/moduledata/live/415.json"
                        onClick={[Function]}
                      />
                    </div>
                  </div>
                </div>
              </ModalBody>
              <ModalFooter>
                <div
                  className="modal-footer"
                >
                  <button
                    className="button-blue action-button test-done-button"
                    onClick={[Function]}
                    type="button"
                  >
                    <WithI18n
                      defaults="Done"
                      id="js.ExportModal.footer.doneButton"
                    >
                      <I18n
                        update={true}
                        withHash={true}
                      >
                        <Trans
                          defaults="Done"
                          i18n={
                            Object {
                              "_": [Function],
                              "_activeLanguageData": Object {
                                "plurals": [Function],
                              },
                              "_activeMessages": Object {},
                              "_catalogs": Object {
                                "en": Object {
                                  "languageData": Object {},
                                  "messages": Object {},
                                },
                              },
                              "_dev": Object {
                                "compile": [Function],
                                "loadLanguageData": [Function],
                              },
                              "_language": "en",
                              "_locales": undefined,
                              "_missing": undefined,
                              "plural": [Function],
                              "select": [Function],
                              "selectOrdinal": [Function],
                              "t": [Function],
                            }
                          }
                          id="js.ExportModal.footer.doneButton"
                        >
                          <Render
                            value="js.ExportModal.footer.doneButton"
                          >
                            js.ExportModal.footer.doneButton
                          </Render>
                        </Trans>
                      </I18n>
                    </WithI18n>
                  </button>
                </div>
              </ModalFooter>
            </div>
          </div>
        </div>
      </Portal>
    </OpenModal>
  </Modal>
</ExportModal>
`;
